extmod/modplatform.c

```diff
@@ -0,0 +1,285 @@
+
+STATIC mp_obj_t platform_version(void) {
+    return MP_OBJ_FROM_PTR(&info_version_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_version_obj, platform_version);
+
+STATIC const qstr platform_uname_fields[] = {
+    MP_QSTR_system,
+    MP_QSTR_node,
+    MP_QSTR_release,
+    MP_QSTR_version,
+    MP_QSTR_machine,
+    MP_QSTR_processor
+};
+
+STATIC MP_DEFINE_ATTRTUPLE(
+    platform_uname_attr_obj,
+    platform_uname_fields,
+    6,
+    MP_ROM_PTR(&info_system_obj),
+    MP_ROM_PTR(&info_node_obj),
+    MP_ROM_PTR(&info_release_obj),
+    MP_ROM_PTR(&info_version_obj),
+    MP_ROM_PTR(&info_machine_obj),
+    MP_ROM_PTR(&info_processor_obj)
+    );
+
+STATIC mp_obj_t platform_uname(void) {
+    return MP_OBJ_FROM_PTR(&platform_uname_attr_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_uname_obj, platform_uname);
+
+STATIC mp_obj_t platform_libc_ver(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    return MP_OBJ_FROM_PTR(&info_libc_tuple_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(platform_libc_ver_obj, 0, platform_libc_ver);
+
+STATIC mp_obj_t platform_hal_ver(void) {
+    return MP_OBJ_FROM_PTR(&info_hal_ver_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_hal_ver_obj, platform_hal_ver);
+
+STATIC const mp_rom_map_elem_t modplatform_globals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_platform) },
+    { MP_ROM_QSTR(MP_QSTR_architecture), MP_ROM_PTR(&platform_architecture_obj) },
+    { MP_ROM_QSTR(MP_QSTR_machine), MP_ROM_PTR(&platform_machine_obj) },
+    { MP_ROM_QSTR(MP_QSTR_node), MP_ROM_PTR(&platform_node_obj) },
+    { MP_ROM_QSTR(MP_QSTR_platform), MP_ROM_PTR(&platform_platform_obj) },
+    { MP_ROM_QSTR(MP_QSTR_processor), MP_ROM_PTR(&platform_processor_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_build), MP_ROM_PTR(&platform_python_build_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_compiler), MP_ROM_PTR(&platform_python_compiler_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_branch), MP_ROM_PTR(&platform_python_branch_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_implementation), MP_ROM_PTR(&platform_python_implementation_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_revision), MP_ROM_PTR(&platform_python_revision_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_version), MP_ROM_PTR(&platform_python_version_obj) },
+    { MP_ROM_QSTR(MP_QSTR_python_version_tuple), MP_ROM_PTR(&platform_python_version_tuple_obj) },
+    { MP_ROM_QSTR(MP_QSTR_release), MP_ROM_PTR(&platform_release_obj) },
+    { MP_ROM_QSTR(MP_QSTR_system), MP_ROM_PTR(&platform_system_obj) },
+    { MP_ROM_QSTR(MP_QSTR_system_alias), MP_ROM_PTR(&platform_system_alias_obj) },
+    { MP_ROM_QSTR(MP_QSTR_version), MP_ROM_PTR(&platform_version_obj) },
+    { MP_ROM_QSTR(MP_QSTR_uname), MP_ROM_PTR(&platform_uname_obj) },
+    { MP_ROM_QSTR(MP_QSTR_libc_ver), MP_ROM_PTR(&platform_libc_ver_obj) },
+    { MP_ROM_QSTR(MP_QSTR_hal_ver), MP_ROM_PTR(&platform_hal_ver_obj) },
```

Comment:
Do we need all of these?  This will add a lot of code size and I'm not sure all of it is necessary.  Which ones are really critical to have?  Maybe there can be the base set, and then extended set includes the rest.

Reply:
No, none of them are necessary, this module provides extra info. That said, I think `system_alias`, `python_version`, `python_branch`, `python_version_tuple` are absolutely useless. The libc, hal, platform, compiler functions on the other hand are the most important ones.